frontend/src/components/GrantAccessForm.vue

```diff
@@ -0,0 +1,183 @@
+                        </div>
+                    </div>
+                </div>
+            </div>
+
+            <div class="row">
+                <div class="col-md-3">
+                    <div class="form-group">
+                        <label for="userIdInput" class="form-label">User Id</label><span class="text-danger">*</span>
+                        <input type="text" id="userIdInput" class="form-control" name="userId" maxlength="20"
+                            placeholder="User's Id"
+                            :class="v$.userId.$dirty ? v$.userId.$invalid ? 'is-invalid' : 'is-valid' : null"
+                            v-model="formData.userId" @blur.native="v$.userId.$touch()">
+                        <div class="text-danger" v-for="error of v$.userId.$errors" :key="error.$uid">
+                            <small>{{ error.$message }}</small>
+                        </div>
+                    </div>
+                </div>
+            </div>
+
+            <div class="row">
+                <div class="col-md-5">
+                    <div class="form-group">
+
+                        <label for="roleSelect" class="control-label">Role</label><span class="text-danger">*</span>
+                        <select id="roleSelect" name="role" class="form-select" aria-label="Role Select"
+                            v-model="formData.role"
+                            :class="v$.role.$dirty ? v$.role.$invalid ? 'is-invalid' : 'is-valid' : null"
+                            @change.native="v$.role.$touch()" @blur.native="v$.role.$touch()">
+                            <option v-for="role in applicationRoleOptions" :value="role">{{ role.role_name }}</option>
+                        </select>
+                        <div class="text-danger" v-for="error of v$.role.$errors" :key="error.$uid">
+                            <small>{{ error.$message }}</small>
+                        </div>
+                    </div>
+                </div>
+            </div>
+
+            <div class="row" v-if="formData.role?.role_name == 'A'">
+                <div class="form-group col-md-3">
+                    <label for="forestClientInput" class="control-label">Forest Client</label>
+                    <input type="text" id="forestClientInput" class="form-control"
+                        :maxlength="FOREST_CLIENT_INPUT_MAX_LENGTH" placeholder="Forest Client Id - 8 digits"
+                        v-model="formData.forestClientNumber" v-on:keypress="onlyDigit($event)">
+                </div>
+            </div>
+
+            <div class="row gy-3">
+                <div class="col-auto">
+                    <button type="submit" id="grantAccessSubmit" class="btn btn-primary mb-3"
+                        :disabled="!v$.$anyDirty || v$.$invalid">
+                        Grant Access
+                    </button>
+                </div>
+            </div>
+        </form>
+    </div>
+</template>
+
+<script lang="ts">
```

Comment:
We're using Vue.js composition API on this project, not options API. Please change this to use composition API.

Reply:
I have changed it to composition API